http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -115,33 +115,37 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         }
       }
 
-      // Create WSM managed identity if shared app
+      wsmResourceApi <- buildWsmResourceApiClient
+
+      // Create or fetch WSM managed identity if shared app
       wsmManagedIdentityOpt <- app.samResourceId.resourceType match {
         case SamResourceType.SharedApp =>
-          childSpan("createWsmIdentityResource").use { implicit ev =>
-            createWsmIdentityResource(app, namespacePrefix, params.workspaceId).map(_.some)
+          // if a managed identity has already been created in the workspace use that otherwise create a new managed identity
+          retrieveWsmManagedIdentity(wsmResourceApi, app.appType, params.workspaceId.value).flatMap {
```

Comment:
TOL, non-blocking: would it be better to pass in the expected identity name to both `retrieve` and `create`, so we are not duplicating the `id<appname>` name schema and introducing a chance they get out of sync?

Reply:
Yeah good point. I will extract that into a function.